### PR TITLE
feat: Support for retrying to deliver SQS message

### PR DESCRIPTION
**Pull request by fredrjoh**
https://linear.app/tibber/issue/INVO-636/retry-when-failing-to-send-email

### CODE REVIEW

**Comment by fredrjoh** [comment]
@naile @danhemma feel free to have a look at this one, I will also add some unit tests to verify that it calls the correct functions at certain scenarios

File: src/queue/QueueSubjectListener.ts [hunk 1]
```diff
@@ -100,28 +119,74 @@ export class QueueSubjectListener {
...
+                    const attempts = parseInt(
+                      attributes?.ApproximateReceiveCount || '1'
+                    );
+
+                    if (attempts < maxAttempts) {
+                      shouldRetry = true;
+                      visibilityTimeout = attempts * backoffDelay;
+
+                      this.logger.debug(
+                        `Message with subject "${m.message.subject}" will be retried`
```
Comment (danhemma) [inline src/queue/QueueSubjectListener.ts:158]: I'd like to include the delay here also

**Review by danhemma — APPROVED** [review]

File: src/queue/QueueSubjectListener.ts [hunk 2]
```diff
@@ -11,12 +14,22 @@ export type QueueSubjectListenerOptions = {
 };
 
 export type QueueSubjectListenerMessageHandler = {
-  (message: unknown, subject: string): void;
+  (message: unknown, subject: string): Promise<void>;
+};
+
+export type QueueSubjectListenerRetryPolicy = {
+  maxAttempts: number;
+  backoffDelay: number;
```
Comment (naile) [inline src/queue/QueueSubjectListener.ts:22]: Can we clarify what this number means, seconds, milliseconds etc?

**Review by naile — COMMENTED** [review]
Looks good. Suggestion: make the retry policy a function instead returning visibility timeout. That way a consumer can have their own backoff logic by changing that function. Ie if someone wants to do less or more aggressive backoff like linear or exponential. The function can also take exception as parameter allowing the consumer to change the logic depending on the error.

**Review by IuryAlves — APPROVED** [review]

**Pull request merged**